android_mic_streaming/app/src/main/java/org/mozilla/deepspeechdemo/MainActivity.kt

```diff
@@ -0,0 +1,137 @@
+package org.mozilla.deepspeechdemo
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.media.AudioFormat
+import android.media.AudioRecord
+import android.media.MediaRecorder
+import android.os.Build
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import kotlinx.android.synthetic.main.activity_main.*
+import org.mozilla.deepspeech.libdeepspeech.DeepSpeechModel
+import org.mozilla.deepspeech.libdeepspeech.DeepSpeechStreamingState
+
+
+class MainActivity : AppCompatActivity() {
+    private var model: DeepSpeechModel? = null
+    private var streamContext: DeepSpeechStreamingState? = null
+
+    private val BEAM_WIDTH = 50
+    private val LM_ALPHA = 0.75f
+    private val LM_BETA = 1.85f
+
+    private val RECORDER_SAMPLERATE = 16000
```

Comment:
We now have API `sampleRate()` entry point for that, please use it :) https://deepspeech.readthedocs.io/en/v0.6.1/Java-API.html#_CPPv4N3org7mozilla10deepspeech13libdeepspeech15DeepSpeechModelE

Reply:
Very nice.